dashboard/modules/job/job_manager.py

```diff
@@ -315,8 +317,11 @@ async def _polling(self, child_process) -> int:
                     await asyncio.sleep(self.SUBPROCESS_POLL_PERIOD_S)
         except Exception:
             if child_process:
-                # TODO (jiaodong): Improve this with SIGTERM then SIGKILL
-                child_process.kill()
+                os.killpg(os.getpgid(child_process.pid), signal.SIGTERM)
+                try:
+                    child_process.wait(self.WAIT_FOR_JOB_TERMINATION_S)
+                except subprocess.TimeoutExpired:
```

Comment:
Could we raise a warning here mentioning that the process group hasn't been killed, so we're killing the child process directly?